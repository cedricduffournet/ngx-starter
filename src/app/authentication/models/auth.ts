export interface AuthToken {
  access_token: string;
  expires_in: number;
  token_type: string;
  refresh_token: string;
}

export interface Credentials {
  login: string;
  password: string;
}

export interface ResetPassword {
  plainPassword: {
    first: string;
    second: string;
  };
}

export interface Register extends ResetPassword {
  civility: number;
  firstname: string;
  lastname: string;
  email: string;
  confirmationUrl: string;
}

export interface SendResetPassword {
  username: string;
}
