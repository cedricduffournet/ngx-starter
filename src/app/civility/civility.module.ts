import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityEffects } from '@app/civility/state/effects';
import {
  CivilityAddComponent,
  CivilityUpdateComponent,
  CivilityDeleteComponent,
  CivilityFormComponent,
  CivilityItemComponent,
  CivilityItemsComponent
} from '@app/civility/components';
import {
  CivilityAddModalComponent,
  CivilityUpdateModalComponent,
  CivilityDeleteModalComponent,
  CivilityListViewComponent
} from '@app/civility/containers';
import { reducers } from '@app/civility/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([CivilityEffects]),
    StoreModule.forFeature('civilities', reducers)
  ],
  declarations: [
    CivilityListViewComponent,
    CivilityAddComponent,
    CivilityUpdateComponent,
    CivilityDeleteComponent,
    CivilityFormComponent,
    CivilityItemComponent,
    CivilityItemsComponent,
    CivilityAddModalComponent,
    CivilityUpdateModalComponent,
    CivilityDeleteModalComponent
  ],
  entryComponents: [
    CivilityUpdateModalComponent,
    CivilityAddModalComponent,
    CivilityDeleteModalComponent
  ],
  exports: [CivilityListViewComponent]
})
export class CivilityModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
