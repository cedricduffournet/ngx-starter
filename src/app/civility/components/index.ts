export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
