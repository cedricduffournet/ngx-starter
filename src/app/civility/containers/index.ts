export * from './civility-add-modal/civility-add-modal.component';
export * from './civility-update-modal/civility-update-modal.component';
export * from './civility-delete-modal/civility-delete-modal.component';
export * from './civility-list-view/civility-list-view.component';
