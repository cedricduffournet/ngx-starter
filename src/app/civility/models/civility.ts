import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
