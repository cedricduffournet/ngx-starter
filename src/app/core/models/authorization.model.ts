export interface Authorization {
  create: boolean;
  update: boolean;
  delete: boolean;
}
