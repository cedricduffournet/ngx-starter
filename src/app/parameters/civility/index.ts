export * from './parameter-civility.component';
