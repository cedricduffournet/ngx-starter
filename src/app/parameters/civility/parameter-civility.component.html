<app-civility-list-view></app-civility-list-view>
