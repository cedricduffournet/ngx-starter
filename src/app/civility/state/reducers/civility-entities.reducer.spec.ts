import { reducer } from '@app/civility/state/reducers/civility-entities.reducer';
import * as fromCivilities from '@app/civility/state/reducers/civility-entities.reducer';

import {
  CivilityListViewActions,
  CivilityUpdateModalActions,
  CivilityApiActions
} from '@app/civility/state/actions';
import { Civility } from '@app/civility/models/civility';

describe('CivilitiyEntitiesReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LOAD_SUCCESS', () => {
    it('should return state with civility entities', () => {
      const civilities = {
        entities: {
          civilities: {
            1: {
              code: 'Mr',
              name: 'Mister'
            },
            2: {
              code: 'Mr',
              name: 'Miss'
            }
          }
        },
        result: [1, 2]
      };
      const action = CivilityApiActions.loadCivilitySuccess({
        civilities
      });
      const result = reducer(fromCivilities.INITIAL_STATE, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('UPDATE', () => {
    const id = 1;
    const civility = {
      code: 'MrUpdated',
      name: 'MisterUpdated'
    } as Civility;
    const data = { id, civility };

    it('should set updating to true', () => {
      const action = CivilityUpdateModalActions.updateCivility({ data });
      const result = reducer(fromCivilities.INITIAL_STATE, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('UPDATE_SUCCESS', () => {
    it('should update civility id 1, upated should be true and updating false', () => {
      const civilities = {
        1: {
          id: 1,
          code: 'Mr',
          name: 'Mister'
        },
        2: {
          id: 2,
          code: 'Mr',
          name: 'Miss'
        }
      };
      const initialState = {
        ...fromCivilities.INITIAL_STATE,
        entities: civilities
      };

      const civility = {
        entities: {
          civilities: {
            1: {
              code: 'MrUpdated',
              name: 'MisterUpdated'
            }
          }
        },
        result: 1
      };
      const action = CivilityApiActions.updateCivilitySuccess({ civility });
      const result = reducer(initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('UPDATE_FAILURE & SHOW_MODAL_UPDATE', () => {
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      updating: true,
      updated: true,
      entities: {
        1: {
          id: 1,
          code: 'Mr',
          name: 'Mister'
        }
      }
    };

    it('should set updating to false', () => {
      const error = {
        message: 'error'
      };
      const action = CivilityApiActions.updateCivilityFailure({ error });
      const result = reducer(initialState, action);

      expect(result).toMatchSnapshot();
    });

    it('should set updating to false, updated to false when show modal', () => {
      const action = CivilityListViewActions.showUpdateCivilityModal();
      const result = reducer(initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('ADD_SUCCESS', () => {
    const civilities = {
      1: {
        id: 1,
        code: 'Mr',
        name: 'Mister'
      },
      2: {
        id: 2,
        code: 'Mr',
        name: 'Miss'
      }
    };
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      entities: civilities
    };

    const civility = {
      entities: {
        civilities: {
          3: {
            code: 'AddCode',
            name: 'AddName'
          }
        }
      },
      result: 3
    };

    it('should add a new civility id 3', () => {
      const action = CivilityApiActions.addCivilitySuccess({ civility });
      const result = reducer(initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('SELECT', () => {
    const civilities = {
      1: {
        id: 1,
        code: 'Mr',
        name: 'Mister'
      },
      2: {
        id: 2,
        code: 'Mr',
        name: 'Miss'
      }
    };
    const initialState = {
      ...fromCivilities.INITIAL_STATE,
      entities: civilities
    };

    const id = 1;

    it('should set selected civility (id 1)', () => {
      const action = CivilityListViewActions.selectCivility({ id });
      const result = reducer(initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    describe('getEntities', () => {
      it('should return entities', () => {
        const result = fromCivilities.getEntities({
          ...fromCivilities.INITIAL_STATE,
          entities: {
            1: {
              id: 1,
              code: 'Mr',
              name: 'Mister'
            },
            2: {
              id: 2,
              code: 'Mr',
              name: 'Miss'
            }
          }
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getSelectedId', () => {
      it('should return selected id (1)', () => {
        const result = fromCivilities.getSelectedId({
          ...fromCivilities.INITIAL_STATE,
          selectedId: 1
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getUpdating', () => {
      it('should return updating true', () => {
        const result = fromCivilities.getUpdating({
          ...fromCivilities.INITIAL_STATE,
          updating: true
        });

        expect(result).toMatchSnapshot();
      });
    });

    describe('getUpdated', () => {
      it('should return updated true', () => {
        const result = fromCivilities.getUpdated({
          ...fromCivilities.INITIAL_STATE,
          updated: true
        });

        expect(result).toMatchSnapshot();
      });
    });
  });
});
