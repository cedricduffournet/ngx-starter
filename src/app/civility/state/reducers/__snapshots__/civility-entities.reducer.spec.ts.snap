// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivilitiyEntitiesReducer ADD_SUCCESS should add a new civility id 3 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "code": "Mr",
      "id": 1,
      "name": "Mister",
    },
    "2": Object {
      "code": "Mr",
      "id": 2,
      "name": "Miss",
    },
    "3": Object {
      "code": "AddCode",
      "name": "AddName",
    },
  },
  "selectedId": null,
  "updated": false,
  "updating": false,
}
`;

exports[`CivilitiyEntitiesReducer LOAD_SUCCESS should return state with civility entities 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "code": "Mr",
      "name": "Mister",
    },
    "2": Object {
      "code": "Mr",
      "name": "Miss",
    },
  },
  "selectedId": null,
  "updated": false,
  "updating": false,
}
`;

exports[`CivilitiyEntitiesReducer SELECT should set selected civility (id 1) 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "code": "Mr",
      "id": 1,
      "name": "Mister",
    },
    "2": Object {
      "code": "Mr",
      "id": 2,
      "name": "Miss",
    },
  },
  "selectedId": 1,
  "updated": false,
  "updating": false,
}
`;

exports[`CivilitiyEntitiesReducer Selectors getEntities should return entities 1`] = `
Object {
  "1": Object {
    "code": "Mr",
    "id": 1,
    "name": "Mister",
  },
  "2": Object {
    "code": "Mr",
    "id": 2,
    "name": "Miss",
  },
}
`;

exports[`CivilitiyEntitiesReducer Selectors getSelectedId should return selected id (1) 1`] = `1`;

exports[`CivilitiyEntitiesReducer Selectors getUpdated should return updated true 1`] = `true`;

exports[`CivilitiyEntitiesReducer Selectors getUpdating should return updating true 1`] = `true`;

exports[`CivilitiyEntitiesReducer UPDATE should set updating to true 1`] = `
Object {
  "entities": Object {},
  "selectedId": null,
  "updated": false,
  "updating": true,
}
`;

exports[`CivilitiyEntitiesReducer UPDATE_FAILURE & SHOW_MODAL_UPDATE should set updating to false 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "code": "Mr",
      "id": 1,
      "name": "Mister",
    },
  },
  "selectedId": null,
  "updated": false,
  "updating": false,
}
`;

exports[`CivilitiyEntitiesReducer UPDATE_FAILURE & SHOW_MODAL_UPDATE should set updating to false, updated to false when show modal 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "code": "Mr",
      "id": 1,
      "name": "Mister",
    },
  },
  "selectedId": null,
  "updated": false,
  "updating": false,
}
`;

exports[`CivilitiyEntitiesReducer UPDATE_SUCCESS should update civility id 1, upated should be true and updating false 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "code": "MrUpdated",
      "name": "MisterUpdated",
    },
    "2": Object {
      "code": "Mr",
      "id": 2,
      "name": "Miss",
    },
  },
  "selectedId": null,
  "updated": true,
  "updating": false,
}
`;

exports[`CivilitiyEntitiesReducer undefined action should return the default state 1`] = `
Object {
  "entities": Object {},
  "selectedId": null,
  "updated": false,
  "updating": false,
}
`;
